File: J2EE_DoANMonHoc_WebTuyenDungViecLam/frontend/src/services/authService.ts
import axiosInstance from './axiosInstance';
import type { ApiResponse, AuthResponse, ChangePasswordRequest, LoginRequest, RegisterRequest, User } from '@/types';

export const authService = {
  login: (data: LoginRequest) =>
    axiosInstance.post<ApiResponse<AuthResponse>>('/auth/login', data),

  register: (data: RegisterRequest) =>
    axiosInstance.post<ApiResponse<AuthResponse>>('/auth/register', data),

  me: () =>
    axiosInstance.get<ApiResponse<User>>('/auth/me'),

  changePassword: (data: ChangePasswordRequest) =>
    axiosInstance.put<ApiResponse<string>>('/auth/change-password', data),
};

File: J2EE_DoANMonHoc_WebTuyenDungViecLam/src/main/java/com/example/DA_WebTuyenDungViecLam/service/AuthService.java
package com.example.DA_WebTuyenDungViecLam.service;

import com.example.DA_WebTuyenDungViecLam.dto.request.LoginRequest;
import com.example.DA_WebTuyenDungViecLam.dto.request.RegisterRequest;
import com.example.DA_WebTuyenDungViecLam.dto.response.AuthResponse;
import com.example.DA_WebTuyenDungViecLam.dto.response.UserResponse;

public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    UserResponse getCurrentUser(String email);
}
